gql/lib/execution.dart

```diff
@@ -199,27 +199,55 @@ class Request {
       );
 }
 
+/// Entry in the [Context].
+///
+/// Any class extending [ContextEntry] may be added to the [Context].
+///
+/// Context entries are a subject to [Request] and [Response] equality comparisons.
+/// Only the values returned by [fieldsForEquality] will be treated as significant.
+@immutable
+abstract class ContextEntry {
+  const ContextEntry();
+
+  /// List of values to be used for equality.
+  List<Object> get fieldsForEquality;
```

Comment:
`equatable` implements a similar strategy for == and hash using mixins. Maybe you can use it instead

Reply:
I do not exactly understand the pros of `equatable`, so I'm hesitant to use it.

Also, I'd like to remind `Link` implementers that they have to think about the `ContextEntry` equality. Exposing this getter will force implementers to think about it. Plus, this way we can transparently switch to `equatable` if needed.